cpp/src/cider/type/plan/StringOpExpr.h

```diff
@@ -367,6 +370,7 @@ class CharLengthStringOper : public StringOper {
       : StringOper(string_oper) {}
 
   std::shared_ptr<Analyzer::Expr> deep_copy() const override;
+  void setIsOutput() override { is_output_ = false; }
```

Comment:
Could you add an argument? It's weird to set it to false.

Reply:
emmm, `is_output_` means whether this expression is the output expression and should this expression do string type output write. for example `concat(substr(col_1, 1, 2), 'abc')`, `concat` should be a output expression and it should do string type output write. While for `charLength(concat(col_1, col2))` , `charLength` is a output expression but it's output is a integer rather than a string, so it should not do extra string write after for loop.

actually, we should have 2 flag. one is is_output_, this should be set via plan side. the other flag is whether return string.

Reply:
I mean `void setIsOutput(bool val)`

Reply:
you mean 
`void StringOper::setIsOutput(bool is_output) { is_output_ = is_output; }`
`void CharLengthStringOper::setIsOutput(bool is_output) { is_output_ = false; }`

Reply:
`CharLengthStringOper` inherit from `StringOper`, which doesn't need to overwrite the `setIsOutput()` definition, just call `setIsOutput(false)`